File: shared/src/commonMain/Subscription.kt
import kotlinx.serialization.Serializable

@Serializable
enum class BillingCycle(val value: String) {
    MONTHLY("monthly"),
    YEARLY("yearly"),
    WEEKLY("weekly")
}

@Serializable
data class Subscription(
    val id: String,
    val userId: String,
    val name: String,
    val price: Double,
    val currency: String,
    val billingCycle: BillingCycle,
    val nextPaymentDate: String, // ISO format date string
    val isActive: Boolean = true
)

File: web-frontend/types/subscription.ts
// Типы для подписок, основанные на shared/src/commonMain/Subscription.kt
export type BillingCycle = 'monthly' | 'yearly' | 'weekly';

export interface Subscription {
  id: string;
  userId: string;
  name: string;
  price: number; // В Kotlin это Double, в TypeScript number
  currency: string;
  billingCycle: BillingCycle;
  nextPaymentDate: string; // ISO format date string
  isActive: boolean;
}

export interface CreateSubscriptionRequest {
  userId: string;
  name: string;
  price: number; // В Kotlin это String, но логичнее использовать number
  currency: string;
  billingCycle: BillingCycle;
  nextPaymentDate: string; // ISO format date string
}